domain/model/state/modelstate.go

```diff
@@ -75,32 +74,32 @@ func (s *ModelState) Delete(ctx context.Context, uuid coremodel.UUID) error {
 	// model being deleted unexpected scenarios.
 	modelTriggerStmt, err := s.Prepare(`DROP TRIGGER IF EXISTS trg_model_immutable_delete;`)
 	if err != nil {
-		return errors.Trace(err)
+		return errors.Capture(err)
 	}
 
 	err = db.Txn(ctx, func(ctx context.Context, tx *sqlair.TX) error {
-		if err := tx.Query(ctx, modelTriggerStmt).Run(); errors.Is(err, sqlair.ErrNoRows) {
-			return modelerrors.NotFound
+		err := tx.Query(ctx, modelTriggerStmt).Run()
+		if errors.Is(err, sqlair.ErrNoRows) {
+			return errors.New("model does not exist").Add(modelerrors.NotFound)
 		} else if err != nil && !internaldatabase.IsErrError(err) {
-			return fmt.Errorf("deleting model trigger %q: %w", uuid, err)
+			return fmt.Errorf("removing model trigger %w", err)
 		}
 
 		var outcome sqlair.Outcome
-		if err := tx.Query(ctx, modelStmt, mUUID).Get(&outcome); err != nil {
-			if errors.Is(err, sql.ErrNoRows) {
-				return modelerrors.NotFound
-			}
-			return fmt.Errorf("deleting model %q: %w", uuid, err)
+		err = tx.Query(ctx, modelStmt, mUUID).Get(&outcome)
+		if err != nil {
+			return errors.Errorf("removing readonly model information: %w", err)
 		}
+
 		if affected, err := outcome.Result().RowsAffected(); err != nil {
-			return fmt.Errorf("deleting model %q: %w", uuid, err)
+			return errors.Errorf("getting result from removing readonly model information: %w", err)
 		} else if affected == 0 {
 			return modelerrors.NotFound
 		}
 		return nil
 	})
 	if err != nil {
-		return errors.Trace(err)
+		return errors.Errorf("deleting model %q from model database: %w", uuid, err)
```

Comment:
If we prefer to use `removing` in the above errors, let's keep it consistent.

Reply:
Done & Done